lib/AST/ASTDemangler.cpp

```diff
@@ -264,25 +264,26 @@ Type ASTBuilder::resolveOpaqueType(NodePointer opaqueDescriptor,
     auto moduleNode = findModuleNode(definingDecl);
     if (!moduleNode)
       return Type();
-    auto parentModule = findModule(moduleNode);
-    if (!parentModule)
+    auto potentialParentModules = findPotentialModules(moduleNode);
+    if (potentialParentModules.empty())
       return Type();
 
-    auto opaqueDecl = parentModule->lookupOpaqueResultType(mangledName);
-    if (!opaqueDecl)
-      return Type();
-    SmallVector<Type, 8> allArgs;
-    for (auto argSet : args) {
-      allArgs.append(argSet.begin(), argSet.end());
-    }
+    for (ModuleDecl *module : potentialParentModules) {
+      auto opaqueDecl = module->lookupOpaqueResultType(mangledName);
+      if (!opaqueDecl)
+        continue;
+      SmallVector<Type, 8> allArgs;
```

Comment:
It's confusing to stick the rest of the body here inside the loop, even though it will only run once.